metrotranscom/doorway
# release: lottery expiration

This PR addresses [#(4058)](https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/gh/bloom-housing/bloom/4058)


## Description

Releases lottery expiration work: https://github.com/bloom-housing/bloom/pull/4184

## How Can This Be Tested/Reviewed?

Setup: add LOTTERY_DAYS_TILL_EXPIRY=45 to your api and partners env files

Sign into Partners Portal and close a listing with a lottery
As an admin, go to the Lottery tab of the closed listing and open the run modal. Verify that expiration statement is present at the top. Run the lottery, releasedToPartners.
As a partner or jurisdictional admin, go to the Lottery tab of the closed listing with a released lottery. Publish the lottery. Verify yellow warning is present at the top of the section with the expiration date included.
In the database, edit the closedAt date on the listing to 45 (or more) days ago. Or change your env variable to 0.
Sign into the Swagger UI page as an admin or jurisdictional admin. Trigger the cron job with the listings/expireLotteries endpoint.
Sign into the Partners Portal as a partner or jurisdictional admin, go to the Lottery tab of the closed listing with a ran lottery. Verify lottery is expired and no data is shown.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
